In [1]:
import pandas as pd

In [38]:
df = pd.read_excel('World_Development_Indicators.xlsx')

In [39]:
df.tail()

Series Name Series Code  \
264                                               NaN         NaN   
265                                               NaN         NaN   
266                                               NaN         NaN   
267  Data from database: World Development Indicators         NaN   
268                          Last Updated: 03/21/2019         NaN   

    Country Name Country Code 1990 [YR1990] 2000 [YR2000] 2009 [YR2009]  \
264          NaN          NaN           NaN           NaN           NaN   
265          NaN          NaN           NaN           NaN           NaN   
266          NaN          NaN           NaN           NaN           NaN   
267          NaN          NaN           NaN           NaN           NaN   
268          NaN          NaN           NaN           NaN           NaN   

    2010 [YR2010] 2011 [YR2011] 2012 [YR2012] 2013 [YR2013] 2014 [YR2014]  \
264           NaN           NaN           NaN           NaN           NaN   
265           NaN           NaN           NaN           NaN           NaN   
266           NaN           NaN           NaN           NaN           NaN   
267           NaN           NaN           NaN           NaN           NaN   
268           NaN           NaN           NaN           NaN           NaN   

    2015 [YR2015] 2016 [YR2016] 2017 [YR2017] 2018 [YR2018]  
264           NaN           NaN           NaN           NaN  
265           NaN           NaN           NaN           NaN  
266           NaN           NaN           NaN           NaN  
267           NaN           NaN           NaN           NaN  
268           NaN           NaN           NaN           NaN

We need to filter the na rows or columns.

* Drop na of 'Series Name'

In [40]:
df.dropna(subset=['Series Name'],axis=0,inplace=True)

In [41]:
df.tail()

Series Name     Series Code  \
261                                 GDP (current US$)  NY.GDP.MKTP.CD   
262                                 GDP (current US$)  NY.GDP.MKTP.CD   
263                                 GDP (current US$)  NY.GDP.MKTP.CD   
267  Data from database: World Development Indicators             NaN   
268                          Last Updated: 03/21/2019             NaN   

    Country Name Country Code 1990 [YR1990] 2000 [YR2000] 2009 [YR2009]  \
261  Yemen, Rep.          YEM   5.64712e+09   9.65244e+09   2.51303e+10   
262       Zambia          ZMB   3.28522e+09   3.60068e+09   1.53283e+10   
263     Zimbabwe          ZWE    8783816700   6.68996e+09   9.66579e+09   
267          NaN          NaN           NaN           NaN           NaN   
268          NaN          NaN           NaN           NaN           NaN   

    2010 [YR2010] 2011 [YR2011] 2012 [YR2012] 2013 [YR2013] 2014 [YR2014]  \
261   3.09067e+10   3.27264e+10   3.54013e+10   4.04152e+10   4.32286e+10   
262   2.02656e+10   2.34601e+10   2.55034e+10   2.80455e+10   2.71506e+10   
263   12041655200   14101920300   1.71148e+10   19091020000   19495519600   
267           NaN           NaN           NaN           NaN           NaN   
268           NaN           NaN           NaN           NaN           NaN   

    2015 [YR2015] 2016 [YR2016] 2017 [YR2017] 2018 [YR2018]  
261   4.55936e+10   3.64367e+10   3.12677e+10            ..  
262   2.11544e+10   2.09548e+10   2.58681e+10            ..  
263   19963120600   2.05487e+10   22040902300            ..  
267           NaN           NaN           NaN           NaN  
268           NaN           NaN           NaN           NaN

The na rows of (264~266) were deleted.

* dropna columns if contain na

In [44]:
df2 = df.dropna(how='any',axis=1)

In [45]:
df2.head()

Series Name
0  GDP (current US$)
1  GDP (current US$)
2  GDP (current US$)
3  GDP (current US$)
4  GDP (current US$)

OK,there are nan values in all columns.